### DISCUSSION TITLE
Replace Font Awesome with alternative icon library.

### DISCUSSION THREAD

**Discussion by anders-swanson** [prometheus/prometheus#10847] [General]
## Proposal
Hi, 
The Font Awesome license bundle includes the SIL OFL 1.1. This license may not be compatible with the legal requirements of some organizations, restricting the use of Prometheus.

It would be ideal to replace Font Awesome entirely with another icon library, such as material-ui (or any permissively licensed library). I was looking through the code base and it looks like Font Awesome is only used for icons, via the Fort Awesome react wrapper - this means replacement is not too difficult of a task.

**Comment by roidelapluie** [comment]
If I am correct, what applies to us is CC-BY 4.0:

```
# Icons: CC BY 4.0 License (https://creativecommons.org/licenses/by/4.0/)

The Font Awesome Free download is licensed under a Creative Commons
Attribution 4.0 International License and applies to all icons packaged
as SVG and JS file types.
```

**Comment by roidelapluie** [comment]
We do not use "icons packaged as web and desktop font files", which are under SIL OFL 1.1, because SVG is directly integrated in the browser.

**Comment by anders-swanson** [comment]
Hi, thanks for replying to my issue.

The SIL OFL 1.1 licensed fonts are still present on the Prometheus server because they are bundled together with the font-awesome NPM package (there is a `fonts` directory containing these bits).

**Comment by caniszczyk** [comment]
In short, pretty sure CC-BY applies here but we aren't going to have an issue with OFL 1.1, however, we need to get it added to the approved list in case this ever pops up again https://github.com/cncf/foundation/issues/360

**Comment by davegandy** [comment]
Hey folks! Creator of Font Awesome here. We've tried to make the license as open as we possibly can. Happy to answer any questions needed to help folks get this sorted.

(And yes, SIL OFL only applies to the typeface itself. If the integration is React, then it's just using SVGs, so the CC-BY license applies. The standard React library includes the attribution automatically in the code, so there's nothing active that should need to be done here as far as I can see.)

**Comment by roidelapluie** [comment]
We are shipping OFL 1.1 items, like fa-solid-900.ttf.gz.

However, it seems a widely accepted license in the industry.

I am interested to know more about 'This license may not be compatible with the legal requirements of some organization', especially if there are real world cases of organizations not using prometheus because of this.

In the meantime, I will convert this into a discussion.

**Comment by anders-swanson** [comment]
Thanks for moving this to a discussion. My organization's legal team has requested us to not ship components that include the SIL OFL 1.1 license.

**Reply by davegandy** [reply]
Can you expand on the reasons why the legal team has issues with SIL OFL? We'd love to understand the issue better.

**Reply by anders-swanson** [reply]
I believe it is a concern with specific wording in the SIL OFL 1.1 which could be considered as copyleft. However I am not a legal expert and cannot comment wholly on the reasoning.

**Reply by davegandy** [reply]
How would we sort through the concern? I'd love to get as concrete as we can so we can have attorneys work through things. We've always attempted to have Font Awesome be as open as we possibly can, and that was why SIL OFL was chosen originally. If this is a sticking point, I'd love to work through it.

**Reply by anders-swanson** [reply]
I can try to get a specific example, but I admit it is cryptic to me.

**Comment by anders-swanson** [comment]
I did find a related issue in AlertManager for Font Awesome: https://github.com/prometheus/alertmanager/issues/1604, suggesting that other organizations have similar legal issues with the licensing.

**Comment by roidelapluie** [comment]
Did you have a look about how we could remove it? Fort awesome seems fine, but the TTF is actually a dependency of bootstrap.

**Comment by anders-swanson** [comment]
> Did you have a look about how we could remove it?

Yes, replacing with [Material UI](https://mui.com/material-ui/icons/#svg-material-icons) (MIT licensed) is not a difficult task. The majority of the FontAwesome icons have a 1:1 similar icon in Material UI.

**Reply by davegandy** [reply]
If the biggest sticking point of FA is that it's not MIT Licensed, we'd consider making a change to FA itself. v1.0 was actually MIT Licensed for the font files and we were told that SIL OFL was more open.

**Reply by JimTheCactus** [reply]
I suspect the clause of interest is (from https://opensource.org/licenses/OFL-1.1):
> 5) The Font Software, modified or unmodified, in part or in whole,
> must be distributed entirely under this license, and must not be
> distributed under any other license. The requirement for fonts to
> remain under this license does not apply to any document created
> using the Font Software.

which is a copyleft clause; derivative works have to *also* use SIL OFL 1.1.

**Reply by anders-swanson** [reply]
Essentially, yes, this is the case. It's a bit of a sticking point for our lawyers.
